src/System.Management.Automation/utils/Verbs.cs

```diff
@@ -1311,6 +1309,107 @@ static Verbs()
+            typeof(VerbsDiagnostic),
+            typeof(VerbsLifecycle),
+            typeof(VerbsOther),
+            typeof(VerbsSecurity)
+        };
+
+        private static string GetVerbGroupDisplayName(Type verbType) => verbType.Name.Substring(5);
+
+        internal static IEnumerable<VerbInfo> FilterByVerbsAndGroups(
+            string[] verbs,
+            string[] groups)
+        {
+            if (groups is null)
+            {
+                foreach (VerbInfo verb in FilterVerbs(verbs))
+                {
+                    yield return verb;
+                }
+
+                yield break;
+            }
+
+            foreach (VerbInfo verb in FilterVerbsWithGroups(verbs, groups))
+            {
+                yield return verb;
+            }
+        }
+
+        private static IEnumerable<VerbInfo> FilterVerbsWithGroups(
+            string[] verbs,
+            string[] groups)
+        {
+            Collection<WildcardPattern> verbPattern = SessionStateUtilities.CreateWildcardsFromStrings(
+                verbs ?? new string[] { "*" },
+                WildcardOptions.IgnoreCase);
+
+            foreach (Type verbType in VerbTypes)
+            {
+                string verbGroupName = GetVerbGroupDisplayName(verbType);
+
+                bool hasVerbGroup = SessionStateUtilities.CollectionContainsValue(
+                    groups,
+                    verbGroupName,
+                    StringComparer.OrdinalIgnoreCase);
+
+                if (hasVerbGroup)
+                {
+                    foreach (VerbInfo verb in FilterVerbsByWildCardPattern(verbType, verbGroupName, verbPattern))
+                    {
+                        yield return verb;
+                    }
+                }
+            }
+        }
+
+        private static IEnumerable<VerbInfo> FilterVerbs(string[] verbs)
+        {
+            Collection<WildcardPattern> verbPattern = SessionStateUtilities.CreateWildcardsFromStrings(
+                verbs ?? new string[] { "*" },
+                WildcardOptions.IgnoreCase);
+
+            foreach (Type verbType in VerbTypes)
+            {
+                string verbGroupName = GetVerbGroupDisplayName(verbType);
+
+                foreach (VerbInfo verb in FilterVerbsByWildCardPattern(verbType, verbGroupName, verbPattern))
+                {
+                    yield return verb;
+                }
+            }
+        }
+
+        private static IEnumerable<VerbInfo> FilterVerbsByWildCardPattern(
+            Type verbType,
+            string verbGroupName,
+            IEnumerable<WildcardPattern> patterns)
+        {
+            foreach (FieldInfo field in verbType.GetFields())
+            {
+                bool matchesWildCardPattern = SessionStateUtilities.MatchesAnyWildcardPattern(
+                    field.Name,
+                    patterns,
+                    defaultValue: false);
+
+                if (field.IsLiteral && matchesWildCardPattern)
```

Comment:
I'd remove `matchesWildCardPattern` helper since it is evaluated before `IsLiteral`.

Reply:
@iSazonov do you mean putting this check after `IsLiteral` check?

Reply:
Actually I think its much more efficient to just match all verbs if `verbs` is nulland remove the default wildcard assignment. Then we can just yield the verb immediately and skip wildcard check.

Reply:
:-) Yes, maybe it would look better if it was like for groups.